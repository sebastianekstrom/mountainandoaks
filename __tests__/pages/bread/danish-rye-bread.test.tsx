import { render, screen } from "@testing-library/react";
import React from "react";

import Danish from "pages/bread/danish-rye-bread";

describe("Danish page", () => {
  it("renders title correctly", () => {
    render(<Danish />);
    const title = screen.queryAllByText(/Danish rye brad/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<Danish />);
    const ingress = screen.queryAllByText(
      /A delicious sweet Danish rye bread that/i,
    );
    expect(ingress).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<Danish />);
    const step = screen.queryAllByText(/Grind the anise and fennel/i);
    expect(step).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<Danish />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });
});
