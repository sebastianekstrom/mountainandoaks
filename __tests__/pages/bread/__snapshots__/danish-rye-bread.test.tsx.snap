// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Danish page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading font-bold text-h3 leading-h3 text-regular "
      >
        Dough
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Filmj√∂lk
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            200
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Beer (stout or porter)
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            150
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Dark syrup
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            133
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Bread flour
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            88
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Rye flour
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            85
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Bicarbonate
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            7
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Anise seeds
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            4
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Fennel seeds
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            4
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Salt
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            3.5
            g
          </p>
        </li>
      </ul>
      <div
        class="ml-auto"
      >
        <button
          class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
          data-testid="BAKERS_PERCENTAGE"
          role="button"
          type="button"
        >
          <p
            class="variable font-sans text-caption leading-paragraph "
          >
            Show bakers %
          </p>
        </button>
      </div>
    </div>
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading font-bold text-h3 leading-h3 text-regular "
      >
        Toppings
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Pumpkin seeds
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Oatmeal
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Sea salt (optional)
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          />
        </li>
      </ul>
    </div>
  </div>
</section>
`;
